presto-bytecode/src/main/java/com/facebook/presto/bytecode/ClassDefinition.java

```diff
@@ -133,7 +140,8 @@ public void visit(ClassVisitor visitor)
         for (int i = 0; i < interfaces.length; i++) {
             interfaces[i] = this.interfaces.get(i).getClassName();
         }
-        visitor.visit(V1_7, toAccessModifier(access) | ACC_SUPER, type.getClassName(), signature, superClass.getClassName(), interfaces);
+        int accessModifier = toAccessModifier(access);
+        visitor.visit(V1_7, isInterface ? accessModifier : accessModifier | ACC_SUPER, type.getClassName(), signature, superClass.getClassName(), interfaces);
```

Comment:
The method call is already complicated, so make this easier to read by doing it the simple way:
```java
int accessModifier = toAccessModifier(access);
if (!isInterface) {
    accessModifier |= ACC_SUPER;
}
```
There are more clever ways, but this is harder to read:
```java
int accessModifier = toAccessModifier(access) | (isInterface ? 0 : ACC_SUPER);